src/highdicom/sr/templates.py

```diff
@@ -385,11 +442,61 @@ def _contains_uidref_items(
     return False
 
 
+def _contains_image_items(
+    parent_item: ContentItem,
+    name: Union[Code, CodedConcept],
+    referenced_sop_class_uid: Union[str, None] = None,
+    referenced_sop_instance_uid: Union[str, None] = None,
+    relationship_type: Optional[RelationshipTypeValues] = None
+) -> bool:
+    """Checks whether an item contains a specific item with value type IMAGE.
+
+    Parameters
+    ----------
+    parent_item: highdicom.sr.ContentItem
+        Parent SR Content Item
+    name: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code]
+        Name of the child SR Content Item
+    referenced_sop_class_uid: Union[str, None], optional
+        SOP Class UID referenced by the content item
+    referenced_sop_instance_uid: Union[str, None], optional
+        SOP Instance UID referenced by the content item
+    relationship_type: Union[highdicom.sr.RelationshipTypeValues, None], optional
+        Relationship between child and parent SR Content Item
+
+    Returns
+    -------
+    bool
+        Whether any of the SR Content Items contained in `parent_item`
+        match the filter criteria
+
+    """  # noqa: E501
+    matched_items = find_content_items(
+        parent_item,
+        name=name,
+        value_type=ValueTypeValues.IMAGE,
+        relationship_type=relationship_type
+    )
+    for item in matched_items:
+        sop_seq = item.ReferencedSOPSequence[0]
+        if referenced_sop_class_uid is not None:
+            if sop_seq.ReferencedSOPClassUID != referenced_sop_class_uid:
+                continue
+        if referenced_sop_instance_uid is not None:
+            if referenced_sop_instance_uid is not None:
+                found_uid = sop_seq.ReferencedSOPInstanceUID
+                if found_uid != referenced_sop_instance_uid:
+                    continue
+        return True
+    return False
+
+
 def _contains_scoord_items(
     parent_item: ContentItem,
     name: Union[Code, CodedConcept],
     graphic_type: Optional[GraphicTypeValues] = None,
-    relationship_type: Optional[RelationshipTypeValues] = None
+    relationship_type: Optional[RelationshipTypeValues] = None,
+    source_image_sop_instance_uid=None
```

Comment:
Type hint is missing

```suggestion
    source_image_sop_instance_uid: Optional[str] = None
```